apps/bbsync/query.py

```diff
@@ -12,6 +14,33 @@
 from .constants import DATE_FMT
 
 
+def summary_shorten(cves, description):
+    """
+    shorten the aligable parts of the tracker summary
+    """
+    # first shorten CVE list
+    if len(cves) > 1:
+
+        # remove the last CVE
+        cves = cves[0:-1]
+        # add the dots to the new last
+        cves[-1] = cves[-1] + " ..."
+
+    # finally shorten the description
+    else:
+        # when we cannot preserve at least a minimal meaningful description
+        # something is fairly wrong and we cannot create such tracker
+        if len(description) <= len(MULTIPLE_DESCRIPTIONS_SUBSTITUTION):
+            raise UnsavableModelError(
+                f"Summary generated for the tracker is longer than {MAX_SUMMARY_LENGTH}"
+            )
```

Comment:
I know this is copied from old code, but we can move this to the top of the function. We avoid doing the operations on the `cves` array and remove this `else` clause making the code cleaner too.

Reply:
I can do it

Reply:
@dmonzonis actually I do not fully understand. How do you mean that we can avoid doing operations on the `cves`? The first mechanism to shorten the tracker summary is removing CVE IDs one at a time. If we avoid it eg. the issue I am trying to solve here (flaw with 35 CVE IDs) will be blocking the sync.

Reply:
What I mean is raising the error first if it needs to be raised. If we do the operations on the `cves` and then raise an error, it won't matter anyway, or will it?

Reply:
That is possible. I was asking about removing the else clause as it does not seem possible to me. I assume that historically there were rarely an issue with the description length so probably therefore first the algorithm deals with CVEs which might be many.

Reply:
Nowadays it can happen that the description is too long as we do not set any restriction there on the title but it is probably still less probably than having too many CVE in case of some multi-CVE tracker

Reply:
I thought the shortening on the `description` was outside of the else clause, I read the indentation wrong. It can stay as is then.